### PR TITLE
HOCS-2563 : add countries and territories files as local resources.

### PR DESCRIPTION
**Pull request by abarker66**
This solution for the removed countries and territories files from the gov.uk registers uses direct copies installed as local resources and served via the rest controller as though it were an external resource, this allows us to us the same object mapping services provided by Spring/Jackson. The application.properties now references the files as local resources.

### CODE REVIEW

File: src/test/resources/application-test.properties
Comment (zuzak) [inline src/test/resources/application-test.properties:25]: We can't hardcode the port number here, as it's variable.

File: src/main/resources/territories-list.json
Comment (zuzak) [inline src/main/resources/territories-list.json:1]: Probably worth running these through a JSON pretty-printer. Everything being on one line will make it very difficult to review changes in future.

File: src/main/java/uk/gov/digital/ho/hocs/info/api/CountryResource.java
Comment (zuzak) [inline src/main/java/uk/gov/digital/ho/hocs/info/api/CountryResource.java:58]: should this be higher than `info`?

File: src/main/resources/territories-list.json
Comment (abarker66) [reply src/main/resources/territories-list.json:1]: we would not review these file but just replace them, they are big and not for humans.

File: src/test/resources/application-test.properties
Comment (abarker66) [reply src/test/resources/application-test.properties:25]: How do I fix that?

**Comment by zuzak** [comment]
We need to check we comply with the terms of the OGL before merging this: we can't just copy/paste code. Might need to attribute inside the `LICENSE` file?

File: src/main/java/uk/gov/digital/ho/hocs/info/api/CountryResource.java
Comment (abarker66) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/CountryResource.java:58]: it is DECS convention, what do you think would be appropriate.

File: src/test/resources/application-test.properties
Comment (abarker66) [reply src/test/resources/application-test.properties:25]: there are a few references to fixed port localhost in this and other properties files - how is this different?

Comment (abarker66) [reply src/test/resources/application-test.properties:25]: I do not know how to fix this. The URL (URI really) is referencing its own Rest Controller, so how do I then reference localhost with the correct port?? do we need some clever config to do this? or should I just reference the file in github like I did in the first solution?

Comment (abarker66) [reply src/test/resources/application-test.properties:25]: I misread this : I have fixed the url with using the server.port variable.

File: src/main/java/uk/gov/digital/ho/hocs/info/api/CountryResource.java
Comment (abarker66) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/CountryResource.java:58]: The failures are now logged at error level. I should point out that there was no logging at all in this class previously.

File: README.md
Comment (zuzak) [inline README.md:126]:
```suggestion
This project contains public sector information licensed under the Open Government Licence v3.0. (http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/)
```

Add this to the LICENSE file too. Just a "Contains public sector information licensed under the Open Government Licence v3.0".

File: src/test/resources/application-test.properties
Comment (LiamMacP) [reply src/test/resources/application-test.properties]: Change to use the port variable have been completed.

File: src/main/java/uk/gov/digital/ho/hocs/info/api/CountryResource.java
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/CountryResource.java]: We do need a review of the the logging levels. I'm happy with this been logged as an error - to which it is.

File: src/main/resources/territories-list.json
Comment (LiamMacP) [reply src/main/resources/territories-list.json]: JSON files have been ran through a prettier - change implemented.

File: LICENSE
Comment (LiamMacP) [inline LICENSE:3]: We should review this - as we use Crown Copyright which is special - I will add a ticket to the backlog to review this.

**Review by LiamMacP — APPROVED** [review]
Changed look good to me. License file have been added and OGL 3.0 reference for the data brought in. Appropriate changes have been made to test properties to allow this to work.

**Pull request merged**
